lib/Data/Validate/Sanctions/Redis.pm

```diff
@@ -50,34 +53,40 @@ sub get_sanctioned_info {
 sub _load_data {
     my $self = shift;
 
-    $self->{last_time}               //= 0;
+    $self->{last_modification}       //= 0;
+    $self->{last_index}              //= 0;
     $self->{_data}                   //= {};
     $self->{_sanctioned_name_tokens} //= {};
     $self->{_token_sanctioned_names} //= {};
 
-    my $last_time = $self->{last_time};
+    return $self->{_data} if $self->{_data} and $self->{last_modification} + $self->IGNORE_OPERATION_INTERVAL > time;
```

Comment:
Here we should to compare not with last modification time but with time when we last time loaded file into memory... we update file once a day... optimization will work only for first 8 minutes after file was updated...